test/async_test.js

```diff
@@ -0,0 +1,165 @@
+/*globals fetch, async */
+"use strict";
+describe("async API tests", () => {
+  it("should asynchronously resolve to the string pass.", () => {
+    var test = async(function* () {
+      var x = yield Promise.resolve("pass");
+      return x;
+    });
+    return test().should.become("pass");
+  });
+
+  it("should reject when rejected.", () => {
+    var error = new Error("testing111");
+    var test = async(function* () {
+      throw error;
+    });
+    return test().should.eventually.be.rejectedWith(error);
+  });
+
+  it("should recover and return in a catch.", () => {
+    var error = new Error("");
+    var test = async(function* () {
+      try {
+        yield Promise.reject(error);
+      } catch (err) {
+        return "pass";
+      }
+    });
+    return test().should.become("pass");
+  });
+
+  it("Should reject after throwing.", () => {
+    var error = new Error("Error");
+    var test = async(function* () {
+      try {
+        yield Promise.reject(error);
+      } catch (err) {
+        throw err;
+      }
+    });
+    return test().should.eventually.be.rejectedWith(error);
+  });
+
+  it("should recover from rejection.", () => {
+    var test = async(function* () {
+      var result = "";
+      try {
+        yield Promise.reject(new Error("pass"));
+      } catch (error) {
+        result = error.message;
+      }
+      return result;
+    });
+    return test().should.become("pass");
+  });
+
+  it("should recover from rejection.", () => {
+    var test = async(function* () {
+      var err = "";
+      try {
+        yield Promise.reject(new Error("exception_"));
+      } catch (ex) {
+        err = ex.message;
+      }
+      var w = yield (err + "recovered_");
+      var z = yield (w + "123");
+      return z; //exception_recovered_123
+    });
+    return test().should.become("exception_recovered_123");
+  });
+
+  it("should recover from rejection, then resolve with a string", () => {
```

Comment:
Yep. Not a big difference. But it passing a promise back uses a different part of the underlying machinery.